extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java

```diff
@@ -187,23 +184,26 @@ void detectBasicAuthImplicitlyRequired(HttpBuildTimeConfig buildTimeConfig,
     }
 
     @BuildStep(onlyIf = IsApplicationBasicAuthRequired.class)
-    AdditionalBeanBuildItem initBasicAuth(HttpBuildTimeConfig buildTimeConfig,
-            BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformerProducer,
+    @Record(ExecutionTime.RUNTIME_INIT)
+    SyntheticBeanBuildItem initBasicAuth(HttpSecurityRecorder recorder,
+            HttpConfiguration runtimeConfig,
+            HttpBuildTimeConfig buildTimeConfig,
             BuildProducer<SecurityInformationBuildItem> securityInformationProducer) {
 
-        if (makeBasicAuthMechDefaultBean(buildTimeConfig)) {
-            //if not explicitly enabled we make this a default bean, so it is the fallback if nothing else is defined
-            annotationsTransformerProducer.produce(new AnnotationsTransformerBuildItem(AnnotationTransformation
-                    .forClasses()
-                    .whenClass(BASIC_AUTH_MECH_NAME)
-                    .transform(ctx -> ctx.add(DefaultBean.class))));
-        }
-
         if (buildTimeConfig.auth.basic.isPresent() && buildTimeConfig.auth.basic.get()) {
             securityInformationProducer.produce(SecurityInformationBuildItem.BASIC());
         }
 
-        return AdditionalBeanBuildItem.builder().setUnremovable().addBeanClass(BasicAuthenticationMechanism.class).build();
+        SyntheticBeanBuildItem.ExtendedBeanConfigurator configurator = SyntheticBeanBuildItem
```

Comment:
I think default bean scope is `@Dependent` and you dropped `@Singleton` so you are making it dependent. It could be actually good idea, but first, I'd need to inspect our code and probably make same change for all the bean. Is this intentional? if not, let's keep it a singleton for now.

Reply:
Yeah, I picked up on that and have the change done locally

Reply:
BTW I glanced at current impl. and I actually think  all the HTTP auth mechanisms would be in 99 % cases better off as dependent (sometimes users can inject them if they provide alternatives, but how many users does it?), I think we just use them once. I'll re-check later this week if I didn't miss something and open PR when Sergey is back.

Reply:
Provided there it is worse to have beans we don't need at CDI container, because these java class instances will live anyway. And with the dependent, we won't be proxying beans either. I am not sure if it is worth it.

Reply:
👍🏽